.github/workflows/sync-branch.yml

```diff
@@ -33,6 +37,15 @@ jobs:
           # add the bin dir of build-cli to the path
           cd packages/build-cli/bin
           echo "$(pwd)" >> $GITHUB_PATH
+
+      # the `url.https://${secret}@github.com/.insteadOf` configuration is used to substitute the default GitHub URL 
+      # with a custom URL that includes a secret token for secure authentication and authorization during Git operations, 
+      # such as pushing to a remote repository on GitHub.
```

Comment:
The docs for this setting are here: https://git-scm.com/docs/git-config#Documentation/git-config.txt-urlltbasegtinsteadOf

And this was helpful for me to understand how these settings look in the git config file: https://gist.github.com/Kovrinic/ea5e7123ab5c97d451804ea222ecd78a